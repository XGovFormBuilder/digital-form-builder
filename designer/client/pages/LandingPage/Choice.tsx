import React, { ReactElement, useState } from "react";
import { i18n } from "../../i18n";
import { withRouter } from "react-router-dom";
import { Radios } from "@govuk-jsx/radios";
import "./LandingPage.scss";

interface Props {
  history: any;
}

export function LandingChoice({ history }: Props): ReactElement {
  const [createNewFrom, setCreateNewForm] = useState(true);

  const handleNext = function () {
    if (createNewFrom) history.push("/new");
    else history.push("/choose-existing");
  };

  const handleChange = (e) => {
    setCreateNewForm(e.target.value == "true");
  };

  return (
    <div className="new-config">
      <div className="choice-wrapper">
        <h1 className="govuk-heading-xl govuk-heading-xl__lowmargin">
          {i18n("landingPage.choice.heading")}
        </h1>
        <p className="govuk-body">{i18n("landingPage.choice.intro")}</p>
        <Radios
          className="govuk-radios--inline"
          name="newOrExisting"
          value={createNewFrom}
          onChange={handleChange}
          required={true}
          fieldset={{
            legend: {
              isPageHeading: true,
              children: [i18n("landingPage.choice.hint")],
            },
          }}
          items={[
            {
              children: [i18n("landingPage.choice.newform")],
              value: true,
            },
            {
              children: [i18n("landingPage.choice.existing")],
              value: false,
            },
          ]}
        />
        <button
          className="govuk-button govuk-button--start"
          onClick={handleNext}
          title="Next"
        >
          {i18n("Next")}
        </button>
      </div>
    </div>
  );
}

export default withRouter(LandingChoice);
