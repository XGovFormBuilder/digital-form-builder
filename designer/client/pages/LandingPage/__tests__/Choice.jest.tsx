import React from "react";
import { LandingChoice } from "../Choice";
import { render, cleanup, fireEvent, screen } from "@testing-library/react";

describe("LandingChoice", () => {
  afterEach(() => jest.resetAllMocks());
  afterEach(cleanup);

  it("snapshot matches", () => {
    const push = jest.fn();
    const history = { push: push };
    const { asFragment } = render(<LandingChoice history={history} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should push /new to history if 'Create a new form' is selected", async () => {
    const push = jest.fn();
    const history = { push: push };
    render(<LandingChoice history={history} />);
    fireEvent.click(screen.getByTitle("Next"));
    expect(push).toBeCalledWith("/new");
  });

  it("should push /choose-existing to history if 'Open an existing form' is selected", async () => {
    const push = jest.fn();
    const history = { push: push };
    render(<LandingChoice history={history} />);
    fireEvent.click(screen.getByLabelText("Open an existing form"));
    fireEvent.click(screen.getByTitle("Next"));
    expect(push).toBeCalledWith("/choose-existing");
  });
});
