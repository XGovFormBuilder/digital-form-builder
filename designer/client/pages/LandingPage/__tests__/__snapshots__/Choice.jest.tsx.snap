// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingChoice snapshot matches 1`] = `
<DocumentFragment>
  <div
    class="new-config"
  >
    <div
      class="choice-wrapper"
    >
      <h1
        class="govuk-heading-xl govuk-heading-xl__lowmargin"
      >
        Design and prototype forms
      </h1>
      <p
        class="govuk-body"
      >
        Use the form designer to easily create,edit forms and test ideas
      </p>
      <div
        class="govuk-form-group "
      >
        <fieldset
          class="govuk-fieldset "
        >
          <legend
            class="govuk-fieldset__legend "
          >
            <h1
              class="govuk-fieldset__heading"
            >
              Select an option
            </h1>
          </legend>
          <div
            class="govuk-radios govuk-radios--inline"
            data-module="govuk-radios"
            required=""
          >
            <div
              class="govuk-radios__item"
            >
              <input
                checked=""
                class="govuk-radios__input"
                id="newOrExisting"
                name="newOrExisting"
                type="radio"
                value="true"
              />
              <label
                class="govuk-label govuk-radios__label "
                for="newOrExisting"
              >
                Create a new form
              </label>
            </div>
            <div
              class="govuk-radios__item"
            >
              <input
                class="govuk-radios__input"
                id="newOrExisting-2"
                name="newOrExisting"
                type="radio"
                value="false"
              />
              <label
                class="govuk-label govuk-radios__label "
                for="newOrExisting-2"
              >
                Open an existing form
              </label>
            </div>
          </div>
        </fieldset>
      </div>
      <button
        class="govuk-button govuk-button--start"
        title="Next"
      >
        Next
      </button>
    </div>
  </div>
</DocumentFragment>
`;
